CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(SugarPP
+        VERSION 0.0.1
+        HOMEPAGE_URL "https://github.com/HO-COOH/SugarPP"
+        LANGUAGES CXX)
+
+# ---- Warning guard ----
+
+# Protect dependents from this project's warnings if the guard isn't disabled
+set(sugarpp_warning_guard "SYSTEM")
+if(SugarPP_INCLUDE_WITHOUT_SYSTEM)
+  set(sugarpp_warning_guard "")
+endif()
+
+# ---- Declare library ----
+
+add_library(SugarPP INTERFACE)
+add_library(SugarPP::SugarPP ALIAS SugarPP)
+
+target_include_directories(SugarPP
+        ${sugarpp_warning_guard}
+        INTERFACE
+        "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
+
+target_compile_features(SugarPP INTERFACE cxx_std_17)
+
+# ---- Install ----
+
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+set(sugarpp_directory "sugarpp-${PROJECT_VERSION}")
+set(sugarpp_include_directory "${CMAKE_INSTALL_INCLUDEDIR}/${sugarpp_directory}")
+
+install(TARGETS SugarPP
+        EXPORT SugarPPTargets
+        INCLUDES DESTINATION "${sugarpp_include_directory}")
+
+set(sugarpp_install_cmakedir
+        "${CMAKE_INSTALL_LIBDIR}/cmake/${sugarpp_directory}")
+
+write_basic_package_version_file(
+        SugarPPConfigVersion.cmake
+        VERSION ${PROJECT_VERSION}
+        COMPATIBILITY SameMajorVersion
+        ARCH_INDEPENDENT)
+
+install(EXPORT SugarPPTargets
+        NAMESPACE SugarPP::
+        DESTINATION "${sugarpp_install_cmakedir}")
+
+install(FILES
+        "${PROJECT_SOURCE_DIR}/cmake/SugarPPConfig.cmake"
+        "${PROJECT_BINARY_DIR}/SugarPPConfigVersion.cmake"
```

Comment:
Probably better to use `${CMAKE_CURRENT_BINARY_DIR}` and `${CMAKE_CURRENT_LIST_DIR}` (same above) instead of the projects source/binary dir.

Reply:
Explanation?

Reply:
If for whatever reason a user uses `include()` instead of `add_subdirectory()` it'll still be correct.

Reply:
That's not a valid reason. `include()` is not for adding directories to the project, it's a glorified copy paste mechanism to inject a CMake file's contents.